src/plugins/plugin.colorschemes.js

```diff
@@ -20,16 +20,30 @@ export default {
 	beforeUpdate: function(chart, options) {
 		var s = options.scheme.split('.');
 		var category = colorschemes[s[0]];
-		var scheme, length, colorIndex, color;
+		var scheme, schemeClone, length, colorIndex, color;
 
 		if (category) {
 			scheme = category[s[1]];
-			length = scheme.length;
+
 			if (scheme) {
+
+				// clone the original scheme
+				schemeClone = scheme.slice();
+
+				// Execute own custom color function
+				var colorFunctionResult = helpers.callback(options.custom, [schemeClone]);
+
+				// check if we really received a filled array; otherwise we keep and use the originally cloned scheme
+				if (colorFunctionResult instanceof Array && colorFunctionResult.length > 0) {
```

Comment:
I prefer `helpers.isArray(colorFunctionResult)` to `colorFunctionResult instanceof Array`.

Reply:
Done.